dbfit-java/hsqldb/src/main/java/dbfit/environment/HSQLDBEnvironment.java

```diff
@@ -65,47 +68,99 @@ protected Pattern getParameterPattern() {
 
     public Map<String, DbParameterAccessor> getAllColumns(String tableOrViewName)
             throws SQLException {
-        String qry = "SELECT COLUMN_NAME, TYPE_NAME FROM INFORMATION_SCHEMA.SYSTEM_COLUMNS\n"
-                + "WHERE TABLE_NAME = ?";
+        String qry = "SELECT column_name"
+                   + "           AS parameter_name"
+                   + "     , type_name"
+                   + "     , 'IN'"
+                   + "           AS parameter_mode"
+                   + "  FROM information_schema.system_columns"
+                   + " WHERE table_name = ?"
+                   + "   AND table_schem = ?";
         return readIntoParams(tableOrViewName, qry);
     }
 
     private Map<String, DbParameterAccessor> readIntoParams(
-            String tableOrViewName, String query) throws SQLException {
+            String objName, String query) throws SQLException {
+        String[] nameParts = objName.toUpperCase().split("\\.");
+        String schemaName = nameParts.length == 2 ? nameParts[0] : getConnection().getSchema();
+        String objectName = nameParts.length == 2 ? nameParts[1] : nameParts[0];
         try (PreparedStatement dc = getConnection().prepareStatement(query)) {
-
-            String tvname;
-            if (tableOrViewName.trim().startsWith("\"") && tableOrViewName.trim().endsWith("\"")) {
+            if (objectName.trim().startsWith("\"") && objectName.trim().endsWith("\"")) {
                 // Remove double quotes.
-                tvname = tableOrViewName.replaceFirst("\"", "");
-                int i = tvname.lastIndexOf("\"");
-                tvname = tvname.substring(0, i) + tvname.substring(i + 1);
-            } else {
-                tvname = tableOrViewName.toUpperCase();
+                objectName = objectName.replaceFirst("\"", "");
+                int i = objectName.lastIndexOf("\"");
+                objectName = objName.substring(0, i) + objName.substring(i + 1);
             }
-
-            dc.setString(1, tvname);
+            dc.setString(1, objectName);
+            dc.setString(2, schemaName);
             ResultSet rs = dc.executeQuery();
             Map<String, DbParameterAccessor> allParams = new HashMap<String, DbParameterAccessor>();
             int position = 0;
             while (rs.next()) {
-                String columnName = rs.getString(1);
-                String dataType = rs.getString(2);
+                String columnName = rs.getString("PARAMETER_NAME");
+                String dataType = rs.getString("TYPE_NAME");
+                Direction direction = null;
+                switch (rs.getString("PARAMETER_MODE")) {
+                    case "IN":
```

Comment:
The values are currently coming from the HSQLDB dictionary views but I can translate them during the query (then parse with `Direction.valueOf()`).